### ISSUE TITLE
Android notifications dismissed prematurely by Chrome extension

### ISSUE THREAD

**Issue by emkael** [joaomgcd/JoinChrome#64]
Hi,

I'm using 1.7.6 version of Chrome extension (Chrome dev channel, v61.0.3135.5) along with 1.7.bf version of the Android app.
Chrome extension is configured with the "Dismiss everywhere" option disabled.

Up until recently, after the timeout defined under "Number of seconds a notification should show", Chrome's notification toast disappeared, but the actual notification (as in Android notification, not Chrome notification) was still available - in the extension popup and, most importantly, on my phone.

For some time (a week?) the notifications began to be dismissed, as if I explicitly clicked the close button on notification toast, as soon as Chrome notification toast is dismissed.

While enabling "Keep notifications until you interact with them" prevents notifications from being unnoticed at all, it's impossible to interact with notifications that expose multiple actions. Once the popup that lists these actions appears, the notification toast is dismissed - and the original notification is dismissed with it, so the popup shows "No notifications".

If it's something specific in the way Chrome started to handle notification toast events, then I guess the "Dismiss everywhere" option should not be optional anymore.
Otherwise, it's some kind of regression.

**Comment by joaomgcd** [comment]
I can't reproduce the issue unfortunately.  Can you please show me how your settings look like under the "Options" and "Advanced" tab? Thanks!

**Comment by emkael** [comment]
In the "Options" tab: only the "Show standard Chrome notification popup" is enabled.

In "Advanced", apart from the settings I've mentioned in the original comment, only "Automatically open any received links" and the Tasker prefix is enabled (though I'm not using Tasker), and I've got some non-default apps added to "App websites" (Messenger Lite, Soundcloud, Friendly).

**Comment by joaomgcd** [comment]
Would it be possible to get a screenshot of those please? :) I just want to make sure I'm using the exact same settings as you. Thanks!

**Comment by emkael** [comment]
Sure. Attached are the screenshots.

"Options" tab, additionally, has no commands nor devices hidden.
And regarding the "Advanced" tab - it's the configuration that works, the notifications are being dismissed as soon as I disable the "Dismiss everywhere" button option.

![join-options](https://user-images.githubusercontent.com/3368944/27582471-4b738316-5b31-11e7-9305-7a6e64eb43cd.png)
![join-advanced](https://user-images.githubusercontent.com/3368944/27582470-4b7227be-5b31-11e7-8061-f2af625b0bf0.png)

**Comment by joaomgcd** [comment]
Thank you very much!
Unfortunately even with those exact settings I could not reproduce the issue...

Maybe it's an issue with the dev channel chrome?

**Comment by emkael** [comment]
> Maybe it's an issue with the dev channel chrome?

Seems like it. Thanks, anyway.

I'll try to poke around a bit, maybe it'll go away and maybe I'll find something.  
Do you want me to keep this issue open, or rather reopen it if I find anything useful?

**Comment by joaomgcd** [comment]
Yes please re-open if you find out it's not a dev channel thing :) Thank you!

**Comment by emkael** [comment]
Re-opening.

Downgraded my Chrome from dev to beta after reporting this issue and it went away.  
Note that this required both a clear Chrome profile (and importing extensions from sync) and re-registering the device in Join.

Now that the beta channel updated to v61 (61.0.3163.39, to be precise), the problem is back.

The moment Chrome notification toast disappears (doesn't matter if the "Dismiss everywhere" command is enabled or not), the notification is dismissed on all devices.

Moreover, if a notification is updated (e.g. mail app notification updates as additional mail arrives), notification actions stop working in a similar fashion to what I've originally mentioned with the "No notifications" message in the popup.

This now happens consistently, on both of my beta Chrome installs (the second one being on beta channel since forever) - so basically forces me to turn off all notification devices in Join.

**Comment by joaomgcd** [comment]
Thanks for the report!
Ok so I tried what you're saying on Chrome Version 61.0.3163.59 and it doesn't behave like that it seems.

My test was:

- Create notification with AutoNotification
- Notification syncs to Chrome
- Notification toast automatically disappears after a few seconds
- Original notification is still present on the phone.

Are you maybe able to try this scenario and see how it works for you? Thanks!

**Comment by peterbonge** [comment]
I've just tested it with AutoNotification and the notification doesn't get removed from the phone, but from the Chrome extension (Chrome Beta). I've attached a screen-vid from what's happening:

![video_2017-08-28_203014](https://user-images.githubusercontent.com/1911329/29788190-035c65fc-8c32-11e7-8f8d-57b7422ce62b.gif)

**Comment by emkael** [comment]
This is exactly the behavior I'm talking about - only if you disable the "Dismiss everywhere" option, the notification disappears as it's dismissed on its disappearance from the extension.

I'll try to provide some more footage later, or maybe you could guide me towards debugging it.

**Comment by joaomgcd** [comment]
Actually I was able to exactly reproduce that now and I found it's a bug in Chrome!

Check this out:

![byuserfalse](https://user-images.githubusercontent.com/4864031/29881987-a435bf24-8da3-11e7-9ba7-16f4c3df8076.PNG)

The **byUser** variable is always set to **true** instead of **false** when the notification is dismissed by Chrome!

I have reported this [here](https://bugs.chromium.org/p/chromium/issues/detail?id=760608) so hopefully they can fix this before the public release. Thanks again!

**Comment by emkael** [comment]
Awesome! <3

Suspected that might be the case, eventually probably would've filed a Chrome bug myself.
